app/src/main/java/com/greenfox/kalendaryo/ChooseAccountActivity.java

```diff
@@ -46,10 +45,12 @@ protected void onCreate(Bundle savedInstanceState) {
 
         String[] array = new String[kalMerged.getInputCalendarIds().size()];
         for (int j = 0; j < kalMerged.getInputCalendarIds().size(); j++) {
-            array[j] = kalMerged.getInputCalendarIds().get(j);
+            array[j] = kalMerged.getInputCalendarIds().get(j).getId();
         }
 
-        kalMerged.setInputCalendarIds(Arrays.asList(array));
+        for (int j = 0; j < kalMerged.getInputCalendarIds().size(); j++) {
+            kalMerged.getInputCalendarIds().get(j).setId(array[j]);
+        }
```

Comment:
@matheb what is this used for? basically here the ids are rewritten for the input calendars